backend/src/roles/roles.guard.ts

```diff
@@ -28,7 +28,8 @@ export class RolesGuard implements CanActivate {
 
     private async getUserRoles(id: string): Promise<Roles[]> {
         const roles = [];
-        if (await this.juniorRepo.findOne(id)) {
+        const isJunior = await this.juniorRepo.findOne(id);
```

Comment:
isJunior implies this is a boolean, even though it is the actual junior object

Reply:
If the item exists, then we will need to carry out the statement beneath. Otherwise we will return undefined, and there fore the condition will fail.